packages/host/app/resources/adoption-chain-manager.ts

```diff
@@ -0,0 +1,104 @@
+import { ImportResource } from './import';
+import { Resource } from 'ember-resources';
+import { tracked } from '@glimmer/tracking';
+import { Loader } from '@cardstack/runtime-common/loader';
+import { getOwner } from '@ember/application';
+import { type BaseDef } from 'https://cardstack.com/base/card-api';
+import type LoaderService from '../services/loader-service';
+import {
+  getCardType,
+  type CardType,
+} from '@cardstack/host/resources/card-type';
+import { restartableTask } from 'ember-concurrency';
+import { TrackedMap } from 'tracked-built-ins';
+
+interface AdoptionChainManagerArgs {
+  named: { importResource: ImportResource | undefined; loader: Loader };
+}
+
+// forwards the card type resource (doesn't care about the card type resource)
+export class AdoptionChainManager extends Resource<AdoptionChainManagerArgs> {
+  importResource: ImportResource | undefined;
+  @tracked cards: (typeof BaseDef)[] = [];
+  @tracked _cardTypes: CardType[] = []; // all cards exported in module
+  ready: Promise<void> | undefined;
+  selections = new TrackedMap<CardType, boolean>();
+
+  modify(_positional: never[], named: AdoptionChainManagerArgs['named']) {
+    this.importResource = named['importResource'];
+    this.ready = this.load.perform();
+  }
+
+  get loading() {
+    return this.load.isRunning;
+  }
+
+  isSelected(cardType: CardType | undefined) {
+    if (cardType === undefined) {
+      return false;
+    }
+    return this.selections.get(cardType) || false;
+  }
+
+  get cardTypes() {
+    return this._cardTypes;
+  }
+
+  get types() {
+    return this._cardTypes.map((t) => t.type);
+  }
+
+  get selectedCardType() {
+    for (let [type, selected] of this.selections.entries()) {
+      if (selected) {
+        return type;
```

Comment:
Is it possible to select multiple cardTypes at one time? If not, I think it is better to make `selections` as an object instead of a map and rename it to `selectedCardType`.

Reply:
Ur right. I dont want to lug this around